strawberry/quart/views.py

```diff
@@ -123,15 +178,58 @@ async def create_streaming_response(
         )
 
     def is_websocket_request(self, request: Request) -> TypeGuard[Request]:
-        return False
+        if has_websocket_context():
+            return True
+
+        # Check if the request is a WebSocket upgrade request
+        connection = request.headers.get("Connection", "").lower()
+        upgrade = request.headers.get("Upgrade", "").lower()
+
+        return "upgrade" in connection and "websocket" in upgrade
 
     async def pick_websocket_subprotocol(self, request: Request) -> Optional[str]:
-        raise NotImplementedError
+        # Get the requested protocols
+        protocols_header = websocket.headers.get("Sec-WebSocket-Protocol", "")
+        if not protocols_header:
+            return None
+
+        # Find the first matching protocol
+        requested_protocols = [p.strip() for p in protocols_header.split(",")]
+        for protocol in requested_protocols:
+            if protocol in self.subscription_protocols:
+                return protocol
+
+        return None
 
     async def create_websocket_response(
         self, request: Request, subprotocol: Optional[str]
     ) -> Response:
-        raise NotImplementedError
+        if subprotocol:
+            # Set the WebSocket protocol if specified
+            await websocket.accept(subprotocol=subprotocol)
+        else:
+            await websocket.accept()
+
+        # Return the current websocket context as the "response"
+        return None
+
+    @classmethod
+    def register_route(cls, app: Quart, rule_name: str, path: str, **kwargs):
+        """Helper method to register both HTTP and WebSocket handlers for a given path.
+
+        Args:
+            app: The Quart application
+            rule_name: The name of the rule
+            path: The path to register the handlers for
+            **kwargs: Parameters to pass to the GraphQLView constructor
+        """
+        # Register both HTTP and WebSocket handler at the same path
+        view_func = cls.as_view(rule_name, **kwargs)
+        app.add_url_rule(path, view_func=view_func, methods=["GET", "POST"])
+
+        # Register the WebSocket handler using the same view function
+        # Quart will handle routing based on the WebSocket upgrade header
+        app.add_url_rule(path, view_func=view_func, methods=["GET"], websocket=True)
```

Comment:
I think I'd prefer if we left it up to the user to register the two URL rules. The examples in our docs for adding HTTP/WS/HTTP+WS URL rules would then end up looking more consistent.